### PR TITLE
8200102: NativeLibraryTest.java fails intermittently, unloaded count is not same as expected

### PR DESCRIPTION
**Pull request by bchristi-git**
Please review this fix for the intermittently-failing java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java.

The change replaces System.gc()+sleep() with the more robust gcAwait() mechanic used elsewhere in the test base, [as pointed out by Martin](https://bugs.openjdk.java.net/browse/JDK-8200102?focusedCommentId=14382648&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14382648)(thanks!).

The new version of the test passes 100 times out of 100 on the test farm.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8200102](https://bugs.openjdk.java.net/browse/JDK-8200102): NativeLibraryTest.java fails intermittently, unloaded count is not same as expected

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**)
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1630/head:pull/1630`
`$ git checkout pull/1630`

### CODE REVIEW

**Comment by mlchung** [comment]
Alternatively, you can simply use `test/libjdk/test/lib/util/ForceGC` in the testlibrary.

**Comment by bchristi-git** [comment]
Yes, I think that's a better plan, Mandy.  Thanks.  I'll update the fix.

**Review by mlchung — APPROVED** [review]

File: test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
```
Comment (naotoj) [inline test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java:2]: Missin 2017?

File: test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java [hunk 2]
```diff
@@ -58,15 +62,14 @@ public static void main(String... args) throws Exception {
...
-            if (count != unloadedCount) {
+            // Unload the class loader and native library, and give the Cleaner
+            // thread a chance to unload the native library.
+            // unloadedCount is incremented when the native library is unloaded.
+            ForceGC gc = new ForceGC();
+            final int finalCount = count;
+            if (!gc.await(() -> finalCount == unloadedCount)) {
                 throw new RuntimeException("Expected unloaded=" + count +
-                    " but got=" + unloadedCount);
+                        " but got=" + unloadedCount);
```
Comment (naotoj) [inline test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java:72]: Could be left unchanged.

**Review by naotoj — APPROVED** [review]
Looks good to me (with trivial nits).

File: test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java (hunk 1 above)
Comment (bchristi-git) [reply test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java:2]: Yes, you're right

File: test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java (hunk 2 above)
Comment (bchristi-git) [reply test/jdk/java/lang/ClassLoader/nativeLibrary/NativeLibraryTest.java:72]: True

**Review by naotoj — APPROVED** [review]
Looks good!

**Review by mlchung — APPROVED** [review]

**Pull request closed without merging**